src/obsloctap/consumekafka.py

```diff
@@ -0,0 +1,184 @@
+# This file is part of obsloctap.
+#
+# Developed for the Rubin Data Management System.
+# This product includes software developed by the Rubin Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# Use of this source code is governed by a 3-clause BSD-style
+# license that can be found in the LICENSE file.
+
+"""Copied from consdb -and perhaps they could be combined in the future"""
+
+import io
+import json
+import math
+import struct
+from typing import Any
+
+import aiokafka
+import httpx
+import structlog
+from aiokafka import ConsumerRecord
+from fastavro import parse_schema, schemaless_reader
+from pandas import DataFrame
+
+from obsloctap.config import config
+from obsloctap.db import DbHelp, DbHelpProvider
+from obsloctap.models import Obsplan
+
+# Configure logging
+log = structlog.getLogger(__name__)
+jaas = ("org.apache.kafka.common.security.scram.ScramLoginModule required",)
+
+# need schedule updates
+topic = "lsst.sal.Scheduler.logevent_predictedSchedule"
+SCHEMA: dict | None = None
+
+COLS = [  # as in the schedule message Kafka or EFD
+    "mjd",
+    "ra",
+    "decl",
+    "exptime",
+    "nexp",
+    "rotSkyPos",
+]
+TO_COLS = [  # as in the ObPlan object
+    "t_planning",
+    "s_ra",
+    "s_dec",
+    "t_exptime",
+    "rubin_nexp",
+    "rubin_rot_sky_pos",
+]
+SINGLE_COLS = [
+    "instrumentConfiguration",
+    "numberOfTargets",
+    "private_efdStamp",
+]
+
+
+def convert_predicted(msg: DataFrame) -> list[Obsplan]:
+    """The predicted schedule msg which is in EFD or comes from Kakfa
+    contains all the columns defined in COLS but all in one row with e.g
+    s_ra_1 s_ra_2. So need to parse that out to someting usefull"""
+
+    plan = []
+    count = 0
+    while f"{COLS[0]}{count}" in msg.columns:
+        p = Obsplan()
+        for cc in range(0, len(COLS)):
+            v = msg[f"{COLS[cc]}{count}"].head().values[0]
+            p.__setattr__(TO_COLS[cc], v)
+            cc += 1
+        p.priority = 2  # more likely than 24 hr schedule
+        if p.t_planning and p.t_planning > 0:
+            plan.append(p)
+        count += 1
+    return plan
```

Comment:
It seems like you might have missed the band for the predicted schedule - instrumentConfiguration has this, but I think there may be a bug. The instrumentConfiguration string is twice as long as the number of targets, as if the bandpass is written down twice for every target in the list. 
I do think the 0 value MJDs are expected targets .. but what their expected time should be is missing. I don't know why some of the visits have a time at all though.